src/kaleidoscope/layers.cpp

```diff
@@ -112,21 +111,22 @@ void Layer_::handleKeymapKeyswitchEvent(Key keymapEntry, uint8_t keyState) {
   }
 }
 
-Key Layer_::eventHandler(Key mappedKey, KeyAddr key_addr, uint8_t keyState) {
+EventHandlerResult Layer_::eventHandler(Key mappedKey, KeyAddr key_addr, uint8_t keyState) {
   if (mappedKey.getFlags() != (SYNTHETIC | SWITCH_TO_KEYMAP))
-    return mappedKey;
+    return EventHandlerResult::OK;
 
   handleKeymapKeyswitchEvent(mappedKey, keyState);
-  return Key_NoKey;
+  return EventHandlerResult::EVENT_CONSUMED;
```

Comment:
This looks suspicious. Is this going to result in layer keys failing to show up in the `active_keys_[]` array? Check `handleKeyswitchEvent()`